File: client/src/enums/appearance/hairColor.ts
/**
 * Represents hair color options for image generation.
 * Used in the User model to store appearance preferences.
 */
export enum HairColor {
  BLACK = "black",
  BROWN = "brown",
  BLONDE = "blonde",
  RED = "red",
  AUBURN = "auburn",
  GRAY = "gray",
  WHITE = "white",
  BALD = "bald",
}

/**
 * Maps hair color values to their human-readable display names
 */
export const HAIR_COLOR_DISPLAY_NAMES: Record<HairColor, string> = {
  [HairColor.BLACK]: "Black",
  [HairColor.BROWN]: "Brown",
  [HairColor.BLONDE]: "Blonde",
  [HairColor.RED]: "Red",
  [HairColor.AUBURN]: "Auburn",
  [HairColor.GRAY]: "Gray",
  [HairColor.WHITE]: "White",
  [HairColor.BALD]: "Bald",
};

File: server/enums/appearance/hair_color.py
from django.db import models


class HairColor(models.TextChoices):
    """
    Enum for hair color options in image generation.
    """

    BLACK = "black", "Black"
    BROWN = "brown", "Brown"
    BLONDE = "blonde", "Blonde"
    RED = "red", "Red"
    AUBURN = "auburn", "Auburn"
    GRAY = "gray", "Gray"
    WHITE = "white", "White"
    BALD = "bald", "Bald"

    def __str__(self) -> str:
        """
        Get a human-readable string representation of the hair color.
        """
        return self.label
